tests/components/tasmota/test_sensor.py

```diff
@@ -342,9 +340,7 @@ async def test_indexed_sensor_state_via_mqtt3(hass, mqtt_mock, setup_tasmota):
     state = hass.states.get("sensor.tasmota_energy_total_1")
     assert state.state == "unavailable"
     assert not state.attributes.get(ATTR_ASSUMED_STATE)
-    assert (
-        state.attributes[sensor.ATTR_STATE_CLASS] == sensor.STATE_CLASS_TOTAL_INCREASING
-    )
+    assert state.attributes[ATTR_STATE_CLASS] is SensorStateClass.TOTAL_INCREASING
```

Comment:
@scop @epenet @frenck I am not sure why `is` works here, but in other cases only `==` works.

Does that mean that some state attributes are written as strings and others as objects?  If so, is that intentional?

I had been assuming based on [this discussion](https://github.com/home-assistant/core/pull/61989#discussion_r770614380) that all state attributes were written as strings and the only way to test them was `==`.

Evidently that's not always the case.

The reason I bring it up: there might be a performance improvement across the whole of HA if there's somewhere a `str` could be traded for a `StrEnum`.

Reply:
Good spot.
I think the main reason is that `state_class` has a single source (`SensorStateClass`) whereas `device_class` has multiple sources (`CoverDeviceClass`, `SensorDeviceClass`, etc.)

Reply:
This PR is for the tasmota tests, not the integration running in Home Assistant. I fail to see relevance of performance with the test suite. `is` is a stronger assertion requiring both variable and expression to be equal in both type and value.

Reply:
@tkdrob `is` is better so no problem with your PR.
The discussion is whether this "discovery" impacts other areas of the core code and/or other PRs.

Reply:
For the PR's I've created, I've added code that actually transforms str values to the appropriate enums so we "know" it will be an enum internally, and added a proper deprecation warning while at it, for example https://github.com/home-assistant/core/pull/60424/files#diff-c49e52963d3481d7a71789cae57190b08c1e283cf64f26bafe0a657574d3d5beR363-R372

I don't know if others are doing this, and it's obviously not this easy if the way to set the values is through direct assignment rather than through a method parameter such as in the case above.

I'm fairly certain that there's no performance impact that would actually matter between equality and identity comparisons for these cases as a whole in HA. Perhaps it's measurable, but at least to me it's no real concern. To me this is about correctness, and IMNSHO identity comparison is the correct way to compare enum members _to each other_. Granted, the cases where equality comparisons produce actually incorrect results are rare, but I'll rather just do the right thing and not guess when the "wrong" way could bite.

Reply:
@scop Perhaps best to continue the discussion in #62225 since this PR is totally fine.

Reply:
Sure, I just replied here because I was summoned/asked here :)